v4/newrelic/transaction.go

```diff
@@ -35,6 +35,23 @@ type thread struct {
 	// isMultiSpan is set to true of the associated transaction has more
 	// than one span or crosses multiple goroutines.
 	isMultiSpan bool
+	logger      Logger
+}
+
+func (thd *thread) logDebug(msg string, context map[string]interface{}) {
```

Comment:
I chose to add these `logDebug` and `logError` methods to the thread, rather than the transaction, because the thread is the common structure shared between `Transaction` and each of the `Segment` types and I wanted to be able to reuse the methods. I feel like it would be cleaner to read to add these methods directly to `Transaction` and each of the segment types, but I didn't want to copy the code so many times.  So to me this was a trade-off between readability and code reuse, and I went for code reuse. Happy to debate this choice, though.